North-Seattle-College/ad440-winter2021-thursday-repo
# Complete implementation of Azure function for /users endpoint - Task 76

Resolves #76 
- This code completes the implementation of the Azure function for the /users endpoint
- The function will be deployed automatically when merged into the development branch
- It doesn't require configuration to be run on the cloud
- If you want to test it locally
  - Make sure you have the Azure functions extension for VSCode
  - Pull the latest iteration of this branch
  - Install the [ODBC driver](https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver15) necessary to connect to the database
  - In local.settings.json at the root of the functions folder, enter in the values for the keys below in the following format: { "Values": { "KEY": "VALUE" } }
    - ENV_DATABASE_NAME (name of the database)
    - ENV_DATABASE_PASSWORD
    - ENV_DATABASE_SERVER (this is the server location)
    - ENV_DATABASE_USERNAME
  - You can click F5 or click run->Start Debugging in the functions folder to start it up
- The implemented function only makes sense for GET and POST requests
  - The GET request doesn't take anything and returns all users
  - The POST request takes a JSON object { "firstName": "a first name", "lastName": "a last name", "email": "an@email.com" } which returns the id of the newly created user in form { "userId": 5 }
  - To test using JSON, copy the link provided by the debugger (the "/users" one) into Postman, and use as usual for GET and POST (info goes into the body)
- All other troubleshooting problems that I know about should be taken care of by error handling but let me know if you run into one.
